### ISSUE TITLE
Angular CLI compatibility 

### ISSUE THREAD

**Issue by Mantisen** [aberezkin/ng2-image-upload#13]
Hi,

I'm using the image uploader in conjunction with an Angular CLI generated project, after importing the component into my app.module.ts, and adding it to a component for testing I'm running into some build issues.

> Error encountered resolving symbol values statically. Calling function 'ImageUploadModule', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol AppModule in..

My app.module.ts looks like this:
```
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { HttpModule } from '@angular/http';
import { RouterModule } from '@angular/router';
import { AlertModule } from 'ng2-bootstrap';
import { ImageUploadModule } from 'angular2-image-upload';

import { AppComponent } from './app.component';
import { ArticlesComponent } from './articles.component';
import { ArticlesService } from './articles.service';
import { ImageService } from './images.service';

import { CarouselModule } from 'ng2-bootstrap';

import { AppRoutingModule } from './app-routing.module';
import { LoginComponent } from './login/login.component';

@NgModule({
  declarations: [
    AppComponent,
    ArticlesComponent,
    LoginComponent
  ],
  imports: [
    BrowserModule,
    FormsModule,
    HttpModule,
    AlertModule.forRoot(),
    AppRoutingModule,
    CarouselModule.forRoot(),
    ImageUploadModule.forRoot()
  ],
  providers: [
    ArticlesService,
    ImageService
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
```

**Comment by aberezkin** [comment]
Hi Mantisen,

I tried to reproduce this error but unsuccessfully. I thought that the root of the problem is `ImageService` because there is a service with the same name in `ImageUploadModule`, but when I added it to my `AppModule` same way as you did I got no errors while building. 

Here's my `AppModule` code:

```
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { HttpModule } from '@angular/http';
import { ImageUploadModule } from 'angular2-image-upload';
import { ImageService } from './image.service';
import { RouterModule } from '@angular/router';

import { AppComponent } from './app.component';
import { AppRoutingComponent } from './app-routing/app-routing.component';
import { AppRoutingModule } from './app-routing/app-routing.module';

@NgModule({
  declarations: [
    AppComponent,
    AppRoutingComponent
  ],
  imports: [
    BrowserModule,
    FormsModule,
    HttpModule,
    RouterModule,
    AppRoutingModule,
    ImageUploadModule.forRoot(),
  ],
  providers: [
    ImageService,
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }

```

First of all, try to rename `ImageSevice` and if this will solve your problem I'll rename `ImageService` in the library to avoid the collision. If this will not help maybe something wrong with the CLI itself, try to update it and build again. 

If nothing helps give me more instructions on how to reproduce the error.

**Comment by AaronLai** [comment]
I got this error too but I did not import image service before . Here is my app module : 

import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { HttpModule } from '@angular/http';

import { AppComponent } from './app.component';
import { MaterialModule } from '@angular/material';
import { Routes, RouterModule } from '@angular/router';
import { MessageComponent } from './message/message.component';
import { HomeComponent } from './home/home.component';
import { CKEditorModule } from 'ng2-ckeditor';

import { ApiConnectorService } from './api-connector.service';
import { RedeemableComponent } from './redeemable/redeemable.component';
import {ImageUploadModule} from "angular2-image-upload";

const appRoutes: Routes = [
    { path: 'home', component: HomeComponent },

  { path: 'message', component: MessageComponent },
    { path: 'redeemable', component: RedeemableComponent }


];


@NgModule({
  declarations: [
    AppComponent,
    MessageComponent,
    HomeComponent,
    RedeemableComponent
  ],
  imports: [
    MaterialModule.forRoot(),
    ImageUploadModule.forRoot(),
    BrowserModule,
    FormsModule,
    HttpModule,
    RouterModule.forRoot(appRoutes),
    CKEditorModule
  ],
  providers: [ApiConnectorService],
  bootstrap: [AppComponent]
})
export class AppModule { }

**Comment by CRISTIANMORALES24131** [comment]
Hi, I have the same problem.

**Comment by kobikd** [comment]
Is there anything new with that?
Got the same error....

**Comment by aberezkin** [comment]
@kobikd Looks like it's bound to the latest angular build. I'll figure out how to solve it in a week.

**Comment by flintcreativestudio** [comment]
Fairly certain this is related: https://github.com/angular/angular/issues/11262

**Comment by flintcreativestudio** [comment]
https://medium.com/@isaacplmann/getting-your-angular-2-library-ready-for-aot-90d1347bcad#.s266g6lnl

**Comment by flintcreativestudio** [comment]
Anyway for all experiencing this issue, its not only with Image Upload module, many other modules are also having the same problem with newer versions of Angular 2 core and CLI.. 
It is best not to use the latest versions of these at the moment, I have restored compatibility to all my modules by using Angular 2.2.1, and CLI version 1.0.0-beta.19-3

Package.json is like this:

> "@angular/common": "~2.2.1",
    "@angular/compiler": "~2.2.1",
    "@angular/core": "~2.2.1",
    "@angular/forms": "~2.2.1",
    "@angular/http": "~2.2.1",
    "@angular/platform-browser": "~2.2.1",
    "@angular/platform-browser-dynamic": "~2.2.1",
    "@angular/router": "~3.1.0",

**Comment by aberezkin** [comment]
@flintcreativestudio 
I got this error while trying to compile with `--aot`

`error TS5023: Unknown compiler option 'aot'.`

**Comment by kobikd** [comment]
Hi
We've tried updating  Angular  and Angular CLI  to the versions that flintcreativestudio said are working,  and we're still running into this problem.
Any ideas?

**Comment by f1k2c** [comment]
any update for this error ?

**Comment by aberezkin** [comment]
@f1k2c I'm thinking about creating an issue in the angular 2 repo to attract ng developers' attention to this problem.

**Comment by caglarturkurka** [comment]
Please help me about this problem because I am waiting solution about this problem

**Comment by pisix** [comment]
Hi, any update about this issue ?

**Comment by VImpe** [comment]
ERROR in Error encountered resolving symbol values statically. Calling function 'ImageUploadModule', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol AppModule in /home/sonu/Angular2/lst-front/src/app/app.module.ts, resolving symbol AppModule in /home/lodo/Angular2/lst-front/src/app/app.module.ts

i got this  errror too

**Comment by aberezkin** [comment]
Resolved by #27. Thanks to our savior @UncleDave.
